### DISCUSSION TITLE
.net6.0-android vs MonoAndroid

### DISCUSSION THREAD

**Discussion by maxkoshevoi** [dotnet/maui#1743] [Q&A] [answered]
Is there a difference between `.net6.0-android` and `MonoAndroid` as target framework?
Do packages that target `MonoAndroid` and not reference any `Xamarin` stuff need to be updated to be used in MAUI?

Example of such package: https://github.com/TheAlmightyBob/Calendars

**Comment by jonathanpeppers** [comment, marked as answer]
> Is there a difference between .net6.0-android and MonoAndroid as target framework?

`MonoAndroid` is Xamarin.Android and `net6.0-android` is .NET 6.

> Do packages that target MonoAndroid and not reference any Xamarin stuff need to be updated to be used in MAUI?

Xamarin.Android NuGet packages will (sometimes?) continue to work fine in .NET 6. As long as they aren't using some API didn't change between the two BCLs. This is similar to migrating from .NET framework to .NET Core.

NuGet packages should start offering `net6.0-android` targets inside -- but they don't necessarily *have to*, if the library is tested on .NET 6 (and still works).

**Reply by maxkoshevoi** [reply]
> This is similar to migrating from .NET framework to .NET Core.

From my understanding NuGets that only target `.NET Framework` cannot be included in `.NET Core` project (even is they only use `NET Standard` APIs), but NuGets that only target `MonoAndroid`, can be included in `net6.0-android` project.

I was interested in how does that work internally?

**Reply by jonathanpeppers** [reply]
You can reference .NET Framework assemblies from a .NET 5 (or higher) at least and you get build warnings.

Any code executed that changed in the BCL, you might hit things like `MissingMethodException`, etc. It just tries to run it and do the best it can.

**Reply by maxkoshevoi** [reply]
Wow, didn't know that, thanks! There's no warning when including MonoAndroid NuGet in net6.0-android project, right? Or maybe I missed it. 
Is that intentional?

**Reply by jonathanpeppers** [reply]
It might be something we add later on, thanks.

**Reply by GalaxiaGuy** [reply]
One related warning, this backwards compatibility is not true of iOS.

`net6.0-ios` will not load `xamarinios10` packages: https://github.com/xamarin/xamarin-macios/issues/13087

**Comment by maxkoshevoi** [comment]
Found an article about it: https://github.com/dotnet/designs/blob/main/accepted/2021/net6.0-tfms/net6.0-tfms.md#design

![image](https://user-images.githubusercontent.com/6609929/148650292-b213dda2-493b-4442-a7a1-cbb409f142a7.png)
